File: benchmarks/tsb/bench_series_floordiv_standalone.ts
/**
 * Benchmark: seriesFloorDiv / seriesMod / seriesPow — standalone floor-division, modulo, and power on 100k Series.
 * Outputs JSON: {"function": "series_floordiv_standalone", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, seriesFloorDiv, seriesMod, seriesPow } from "../../src/index.ts";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => (i % 1000) + 1) });

for (let i = 0; i < WARMUP; i++) {
  seriesFloorDiv(s, 3);
  seriesMod(s, 7);
  seriesPow(s, 2);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesFloorDiv(s, 3);
  seriesMod(s, 7);
  seriesPow(s, 2);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_floordiv_standalone",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_series_floordiv_standalone.py
"""
Benchmark: Series floordiv / mod / pow standalone functions on 100k Series.
Mirrors seriesFloorDiv / seriesMod / seriesPow.
Outputs JSON: {"function": "series_floordiv_standalone", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

data = (np.arange(SIZE) % 1000) + 1
s = pd.Series(data.astype(float))

for _ in range(WARMUP):
    s.floordiv(3)
    s.mod(7)
    s.pow(2)

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.floordiv(3)
    s.mod(7)
    s.pow(2)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_floordiv_standalone",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
